fix(server): strip metadata from thumbnails

uhthomas opened the pull request:
#3658 introduced support for thumbnail ICC profiles, but also inadvertently included all thumbnail metadata. It seems this has to be explicitly disabled.

Refs: #4382

mertalev reviewed (changes requested):
I don't think false is a valid argument here for the withMetadata API. The second call also overwrites the first; they can't be stacked like that.

uhthomas commented:
> I don't think false is a valid argument here for the withMetadata API. The second call also overwrites the first; they can't be stacked like that.

It is valid, but missing from the TypeScript definition (https://github.com/lovell/sharp/issues/3822).

I think you're right though, and the original change would not behave as expected.

https://github.com/lovell/sharp/blob/eefaa998725cf345227d94b40615e090495c6d09/src/pipeline.cc#L792

I've updated the PR to first create the resized image without metadata, and then a second image with the ICC profile.

mertalev reviewed (commented):
I think this looks good, but it needs testing to make sure it handles colors and bit depth correctly.

jrasm91 commented:
Can you add to the description how you have tested this? What formats you validated it with, etc.

uhthomas committed:
fix(server): strip metadata from thumbnails

included all thumbnail metadata. It seems this has to be explicitly disabled.

Refs: #4382

feat. basic metadata e2e test

fix: use tiff thumbnails in first step + e2e fix

fix: revert switch to tiff

feat: test metadata of both webp and jpg

feat: use upload in e2e test

fix: lint

strip metadata with exiftool

use `withIccProfile`

fix e2e

formatting

run jobs in e2e

alextran1502 commented:
@etnoy When you have a chance can you help with fixing the e2e test?

etnoy commented:
> @etnoy When you have a chance can you help with fixing the e2e test?

should be done now!

yodatak commented:
I think i got an error too i cannot make any thumbernails if i have PREFER WIDE GAMUT enabled

mertalev commented:
I updated this branch to use the new sharp feature to set ICC profiles and rebased. Tested by right-clicking a preview image and confirming EXIF metadata is stripped and the ICC profile is set.
